imqsauth/http.go

```diff
@@ -487,6 +515,45 @@ func httpHandlerCreateUser(central *ImqsCentral, w http.ResponseWriter, r *httpR
 	}
 }
 
+func httpHandlerUpdateUser(central *ImqsCentral, w http.ResponseWriter, r *httpRequest) {
+	strUserId := strings.TrimSpace(r.http.URL.Query().Get("userid"))
+	email := strings.TrimSpace(r.http.URL.Query().Get("email"))
+	username := strings.TrimSpace(r.http.URL.Query().Get("username"))
+	firstname := strings.TrimSpace(r.http.URL.Query().Get("firstname"))
+	lastname := strings.TrimSpace(r.http.URL.Query().Get("lastname"))
+	mobilenumber := strings.TrimSpace(r.http.URL.Query().Get("mobilenumber"))
+
+	userId, err := strconv.ParseInt(strUserId, 10, 64)
+	if err != nil {
+		authaus.HttpSendTxt(w, http.StatusBadRequest, err.Error())
+	}
+	if err := central.Central.UpdateIdentity(authaus.UserId(int64(userId)), email, username, firstname, lastname, mobilenumber); err != nil {
+		authaus.HttpSendTxt(w, http.StatusForbidden, err.Error())
+	} else {
+		authaus.HttpSendTxt(w, http.StatusOK, fmt.Sprintf("Updated user: '%v'", strUserId))
+	}
+}
+
+func httpHandlerArchiveUser(central *ImqsCentral, w http.ResponseWriter, r *httpRequest) {
+	strUserId := strings.TrimSpace(r.http.URL.Query().Get("userid"))
+	userId, err := strconv.ParseInt(strUserId, 10, 64)
+	if err != nil {
+		authaus.HttpSendTxt(w, http.StatusBadRequest, err.Error())
+		return
+	}
+	// admin permission allows you to archive a user account
+	if authaus.UserId(userId) != r.token.UserId && !r.permList.Has(PermAdmin) {
```

Comment:
Should we really allow a person to archive their own account? It just seems like the kind of thing prone to accident - probably better to only allow an admin to do this.